exercises/yacht/canonical-data.json

```diff
@@ -0,0 +1,242 @@
+            "property":"score",
+            "input":{
+                "dice":[3, 3, 3, 3, 3],
+                "category":"fives"
+            },
+            "expected":0
+        },
+        {
+            "description":"Sixes",
+            "property":"score",
+            "input":{
+                "dice":[2, 3, 4, 5 ,6],
+                "category":"sixes"
+            },
+            "expected":6
+        },
+        {
+            "description":"Full house two small, three big",
+            "property":"score",
+            "input":{
+                "dice":[2, 2, 4, 4, 4],
+                "category":"full house"
+            },
+            "expected":16
+        },
+        {
+            "description":"Full house three small, two big",
+            "property":"score",
+            "input":{
+                "dice":[5, 3, 3, 5, 3],
+                "category":"full house"
+            },
+            "expected":19
+        },
+        {
+            "description":"Two pair is not a full house",
+            "property":"score",
+            "input":{
+                "dice":[2, 2, 4, 4, 5],
+                "category":"full house"
+            },
+            "expected":0
+        },
+        {
+            "description":"Yacht is not a full house",
+            "property":"score",
+            "input":{
+                "dice":[2, 2, 2, 2, 2],
+                "category":"full house"
+            },
+            "expected":0
+        },
+        {
+            "description":"Four of a Kind",
+            "property":"score",
+            "input":{
+                "dice":[6, 6, 4, 6, 6],
+                "category":"four of a kind"
+            },
+            "expected":24
+        },
+        {
+            "description":"Yacht can be scored as Four of a Kind",
+            "property":"score",
+            "input":{
+                "dice":[3, 3, 3, 3, 3],
+                "category":"four of a kind"
+            },
+            "expected":12
+        },
+        {
+            "description":"Full house is not Four of a Kind",
+            "property":"score",
+            "input":{
+                "dice":[3, 3, 3, 5, 5],
+                "category":"four of a kind"
+            },
+            "expected":0
+        },
+        {
+            "description":"Little Straight",
+            "property":"score",
+            "input":{
+                "dice":[3, 5, 4, 1, 2],
+                "category":"little straight"
+            },
+            "expected":30
+        },
+        {
+            "description":"Little Straight as Big Straight",
+            "property":"score",
+            "input":{
+                "dice":[1, 2, 3, 4, 5],
+                "category":"big straight"
+            },
+            "expected":0
+        },
+        {
+            "description":"Four in order but not a little straight",
+            "property":"score",
+            "input":{
+                "dice":[1, 1, 2, 3, 4],
```

Comment:
1, 1, 2, 3, 4 is an interesting case (so I think it should be kept?), but I had a different case in mind: 1, 1, 2, 3, 5. An implementation may just ask the question: is the max five and the min 1? but that implementation would be mistaken here.

Reply:
I see what you're getting at.  I've added that as well.